packages/foundation-models/README.md

```diff
@@ -17,9 +17,153 @@ $ npm install @sap-ai-sdk/foundation-models
+
+### Prerequisites
+
+- A deployed OpenAI model in SAP generative AI hub.
+  - You can use the [`DeploymentApi`](../ai-api/README.md#deploymentapi) from `@sap-ai-sdk/ai-api` to deploy a model to SAP generative AI hub. For more information, see [here](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-deployment-for-generative-ai-model-in-sap-ai-core)
+- `sap-ai-sdk/foundation-models` package installed in your project.
+
+### Usage of Azure OpenAI Chat Client
+
+Use the `AzureOpenAiChatClient` to send chat completion requests to an OpenAI model deployed in SAP generative AI hub.
+You can pass the model name as a parameter to the client, the sdk will implicitly fetch the deployment ID for the model from the AI Core service and use it to send the request.
+
+The deployment information which includes deployment ID and properties like model name and model version is also cached by default for 5 mins so that performance is not impacted by fetching the deployment information for every request.
+
+```TS
+import { AzureOpenAiChatClient } from '@sap-ai-sdk/foundation-models';
+
+const client = new AzureOpenAiChatClient('gpt-35-turbo');
+const response = await client.run({
+  messages: [
+    {
+      role: 'user',
+      content: 'Where is the deepest place on earth located'
+    }
+  ]
+});
+
+const responseContent = response.getContent();
+
+```
+
+To send a chat completion request with system messages use:
+
+```TS
+import { AzureOpenAiChatClient } from '@sap-ai-sdk/foundation-models';
+
+const client = new AzureOpenAiChatClient('gpt-35-turbo');
+const response = await client.run({
+  messages: [
+    {
+      role: 'system',
+      content: 'You are a friendly chatbot.'
+    },
+    {
+      role: 'user',
+      content: 'Hi, my name is Isa'
+    },
+    {
+      role: 'system',
+      content: 'Hi Isa! It is nice to meet you. Is there anything I can help you with today?'
+    },
+    {
+      role: 'user',
+      content: 'Can you remind me, What is my name?'
+    }
+  ],
+  max_tokens: 100,
+  temperature: 0.0
+});
+
+const responseContent = response.getContent();
+const tokenUsage = response.getTokenUsage();
+
+logger.info(
+  `Total tokens consumed by the request: ${tokenUsage.total_tokens}\n` +
+  `Input prompt tokens consumed: ${tokenUsage.prompt_tokens}\n` +
+  `Output text completion tokens consumed: ${tokenUsage.completion_tokens}\n`
+);
+
+```
+
+You can see that one can send multiple messages in a single request.
+This is useful in providing a history of the conversation to the model.
+
+Some parameters like `max_tokens` and `temperature` can be also be passed to the request to control the completion behavior.
+Refer to `AzureOpenAiChatCompletionParameters` interface for knowing more parameters that can be passed to the chat completion request.
+
+#### Obtaining a client using deployment ID
```

Comment:
[pp] I'd change this to focus on resource group instead, maybe even leave out deployment ID completely